Brave Beta/Developer  fails to launch on Windows

<!--
Have you searched for similar issues? We have received a lot of feedback and bug reports that we have closed as duplicates. Before submitting this issue, please visit our community site for common ones: https://community.brave.com/c/common-issues
-->

### Description
Brave Beta/DEveloper fails to launch on Windows

### Steps to Reproduce
  1. Install Brave beta 0.20.5 or Brave developer 0.21.2 along side 0.19.80
  2. Browser launches on completion of setup
  3. Quit both browsers Beta and release version
  4. Launch Brave beta/developer, doesn't launch background worker process for Brave beta/developer keeps initialising and terminates, Can't terminate the background process(see screenshot)
  5. Launch Brave, still doesn't launch background worker process for Brave beta/developer keeps initialising and terminating


**Actual result:**
![bravebeta](https://user-images.githubusercontent.com/17010094/32413334-463c6182-c207-11e7-8eb1-44d4fb0b7c5c.gif)

**Expected result:**
Brave beta/developer browser window should launch 

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**


**Reproducible on current live release:**
N/A

### Additional Information
cc: @bbondy @aekeus

This happens on building from source as well. Browser window doesn't start. Console throws this error
```
$ npm start

> brave@0.20.5 start E:\srirambv\master\browser-laptop
> node ./tools/start.js --user-data-dir-name=brave-development --enable-logging --v=0 --enable-extension-activity-logging --enable-sandbox-logging --enable-dcheck

App threw an error during load
Error: Cannot find module '../auto-updater'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (E:\srirambv\master\browser-laptop\node_modules\electron-prebuilt\dist\resources\electron.asar\common\reset-search-paths.js:41:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.get [as autoUpdater] (E:\srirambv\master\browser-laptop\node_modules\electron-prebuilt\dist\resources\electron.asar\browser\api\exports\electron.js:17:14)
    at E:\srirambv\master\browser-laptop\app\updater.js:10:30
    at Object.<anonymous> (E:\srirambv\master\browser-laptop\app\updater.js:258:2)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
An uncaught exception occurred in the main process Uncaught Exception:
Error: Cannot find module '../auto-updater'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (E:\srirambv\master\browser-laptop\node_modules\electron-prebuilt\dist\resources\electron.asar\common\reset-search-paths.js:41:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.get [as autoUpdater] (E:\srirambv\master\browser-laptop\node_modules\electron-prebuilt\dist\resources\electron.asar\browser\api\exports\electron.js:17:14)
    at E:\srirambv\master\browser-laptop\app\updater.js:10:30
    at Object.<anonymous> (E:\srirambv\master\browser-laptop\app\updater.js:258:2)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
[1106/154045.171:ERROR:process_info.cc(632)] range at 0xbd05be8200000000, size 0x275 fully unreadable
[1106/154045.172:ERROR:process_info.cc(632)] range at 0xbd05bea200000000, size 0x275 fully unreadable
[1106/154045.172:ERROR:process_info.cc(632)] range at 0xbd05bea400000000, size 0x275 fully unreadable
Waiting 60 seconds for process to load
Error: unrecognized flag --icu_case_mapping
Try --help for options
process exited with code 0
```

--- a/app/windowsInit.js
+++ b/app/windowsInit.js
@@ -75,8 +75,8 @@ if (process.platform === 'win32') {
     process.exit(0)
   }
 
-  const userDataDirSwitch = '--user-data-dir=brave-' + channel
-  if (channel !== 'dev' && !process.argv.includes(userDataDirSwitch)) {
+  const userDataDirSwitch = '--user-data-dir-name=brave-' + channel
+  if (channel && channel !== 'dev' && !process.argv.includes(userDataDirSwitch)) {
     if (cmd === '--squirrel-firstrun') {
       app.relaunch({args: [userDataDirSwitch, '--relaunch']})
     } else {

